File: services/node/src/utils/mailer.js
const logger = require('../logger');

async function sendEmail({ to, subject, text, html, attachments = [] }) {
  // Stub mailer: log and resolve. Replace with real provider (e.g., SES, SendGrid) later.
  logger.info({ to, subject, attachmentsCount: attachments.length }, 'Sending email (stub)');
  return { ok: true };
}

module.exports = { sendEmail };

File: services/python/edtech_service/utils/mailer.py
import structlog

logger = structlog.get_logger()

def send_email(*, to: str, subject: str, text: str = "", html: str = "", attachments: list | None = None) -> dict:
    attachments = attachments or []
    logger.info("sending_email_stub", to=to, subject=subject, attachments_count=len(attachments))
    return {"ok": True}
